janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/step/JanusGraphPropertiesStep.java

```diff
@@ -45,23 +45,36 @@
  */
 public class JanusGraphPropertiesStep<E> extends PropertiesStep<E> implements HasStepFolder<Element, E>, Profiling, MultiQueriable<Element,E> {
 
-    private boolean initialized = false;
     private boolean useMultiQuery = false;
+    private Set<Vertex> verticesToPrefetch = new HashSet<>();
     private Map<JanusGraphVertex, Iterable<? extends JanusGraphProperty>> multiQueryResults = null;
     private QueryProfiler queryProfiler = QueryProfiler.NO_OP;
 
     public JanusGraphPropertiesStep(PropertiesStep<E> originalStep) {
         super(originalStep.getTraversal(), originalStep.getReturnType(), originalStep.getPropertyKeys());
         originalStep.getLabels().forEach(this::addLabel);
-        this.hasContainers = new ArrayList<>();
-        this.limit = Query.NO_LIMIT;
+
+        if (originalStep instanceof JanusGraphPropertiesStep) {
```

Comment:
This can definitely be true with the old traversal optimization order of TP < 3.5, but I'm not sure if that is still the case in the current version.
This condition occurs if there is a `JanusGraphPropertiesStep` present in a traversal before `JanusGraphLocalQueryOptimizerStrategy` is executed on the traversal. This strategy replaces all `PropertiesStep`s with `JanusGraphPropertyStep`s. And since an already existing `JanusGraphPropertyStep` is still a `PropertiesStep`, it will be replaced as well.